### PR TITLE
replace index.js entry point

### PR DESCRIPTION
**Pull request by sachinraja**
follow up from #6789

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -1,24 +0,0 @@
-export interface SimpleIcon {
```
Comment (mondeja) [inline index.d.ts:1]: This file should not be renamed to `types.d.ts` instead of be removed?

Comment (sachinraja) [reply index.d.ts:1]: good catch, fixed that

**Review by mondeja — APPROVED** [review]
Looks good, thanks @sachinraja! 🥇

**Review by mondeja — APPROVED** [review]
Just made some changes, now really looks good 😅

**Pull request merged**